cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -170,16 +170,18 @@ public void apply() throws Exception {
     }
   }
 
-  private void executePersist(final TriggerKey triggerKey, final Trigger.TriggerState state) {
+  private void executePersist(final TriggerKey triggerKey, final Trigger.TriggerState newTriggerState) {
     try {
       factory.createExecutor(ImmutableList.of((TransactionAware) table))
                             .execute(new TransactionExecutor.Subroutine() {
                               @Override
                               public void apply() throws Exception {
                                 if (triggerKey != null) {
-                                  TriggerStatus trigger = readTrigger(triggerKey);
-                                  if (trigger != null) {
-                                    persistTrigger(table, trigger.trigger, state);
+                                  TriggerStatus storedTriggerStatus = readTrigger(triggerKey);
+                                  if (storedTriggerStatus != null) {
```

Comment:
So, if this is called with a null triggerKey, that would be an internal error? Shouldn't we throw an exception then?

And if it is called for a non-existent trigger, also? Or at least log a warning, because we do not expect this to happen.